gmacros: add G_GNUC_UNUSED to G_STATIC_ASSERT
Silence some warnings seen with GCC 4.8.


author	Cosimo Cecchi <cosimoc@gnome.org>	
committer	Cosimo Cecchi <cosimoc@gnome.org>
commit	0ea7abaed5cff685f4d825f13588979e5dcb8e5b (patch)
tree	42be0fa62f2a2330dd59e418254b88a26c698f26
parent	7986d834c03490104a06e6adfa6156a52688f7c2 (diff)
diff --git a/glib/gmacros.h b/glib/gmacros.h
index b07eba7..7c21717 100644
--- a/glib/gmacros.h
+++ b/glib/gmacros.h
@@ -159,9 +159,9 @@
 #define G_PASTE_ARGS(identifier1,identifier2) identifier1 ## identifier2
 #define G_PASTE(identifier1,identifier2)      G_PASTE_ARGS (identifier1, identifier2)
 #ifdef __COUNTER__
-#define G_STATIC_ASSERT(expr) typedef char G_PASTE (_GStaticAssertCompileTimeAssertion_, __COUNTER__)[(expr) ? 1 : -1]
+#define G_STATIC_ASSERT(expr) typedef char G_PASTE (_GStaticAssertCompileTimeAssertion_, __COUNTER__)[(expr) ? 1 : -1] G_GNUC_UNUSED
 #else
-#define G_STATIC_ASSERT(expr) typedef char G_PASTE (_GStaticAssertCompileTimeAssertion_, __LINE__)[(expr) ? 1 : -1]
+#define G_STATIC_ASSERT(expr) typedef char G_PASTE (_GStaticAssertCompileTimeAssertion_, __LINE__)[(expr) ? 1 : -1] G_GNUC_UNUSED
 #endif
 #define G_STATIC_ASSERT_EXPR(expr) ((void) sizeof (char[(expr) ? 1 : -1]))
 #endif

